README.md
# RumahSakit
Tugas Akhir UAS Pemrograman Berbasis Objek


changelog.md
/------------------------------------------------------------- RumahSakit ------------------------------------------------------------/

> > > Program name : RumahSakit
> > > Year : 2021-now
> > > Type : Java File
> > > Contributor : Argha, Guntur, Irfan, Naufal, Rayhan 


(All notable changes to this project will be documented in this file.)

Format:
dd MM YYY - [Author]

Write every changes for the version in here

-------------------------------------------------------------/

**[Write_new/latest_changes_here]**

**05 June 2021 - [Rayhan] **

Added
- Initial Files


**05 June 2021 - [Guntur] **

Added
- Created on new, update, delete and save button also with the function/procedure
- Passing data table to form

**06 June 2021 - [Guntur] **

Changed
- DataPasien connected to db

**06 June 2021 - [Guntur] **

Changed
- DataPasien & DataDokter variable function

Added
- DataPasien & DataDokter comments 

**06 June 2021 - [Rayhan] **

Added
- Invoice.java

**07 June 2021 - [Rayhan] **

Fixed
- Invoice.java worked


RumahSakit/src/rumahsakit/login.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package rumahsakit;

import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.Statement;
import javax.swing.JOptionPane;

/**
 *
 * @author irfannm
 */
interface a{
    
};
public class login extends javax.swing.JFrame {

    /**
     * Creates new form login
     */
    ResultSet rs;
    String sql;
    public login() {
        
        initComponents();
    }

    /**
     * This method is called from within the constructor to initialize the form.
     * WARNING: Do NOT modify this code. The content of this method is always
     * regenerated by the Form Editor.
     */
    @SuppressWarnings("unchecked")
    // <editor-fold defaultstate="collapsed" desc="Generated Code">//GEN-BEGIN:initComponents
    private void initComponents() {

        jLabel4 = new javax.swing.JLabel();
        jLabel5 = new javax.swing.JLabel();
        inputUsername = new javax.swing.JTextField();
        jLabel6 = new javax.swing.JLabel();
        buttonMasuk = new javax.swing.JButton();
        inputPassword = new javax.swing.JPasswordField();
        jLabel2 = new javax.swing.JLabel();
        jLabel1 = new javax.swing.JLabel();

        setDefaultCloseOperation(javax.swing.WindowConstants.EXIT_ON_CLOSE);
        setTitle("Maju Jaya Medical");
        setMinimumSize(new java.awt.Dimension(666, 460));
        setPreferredSize(new java.awt.Dimension(666, 460));
        getContentPane().setLayout(null);

        jLabel4.setForeground(new java.awt.Color(1, 1, 1));
        jLabel4.setText("Username");
        getContentPane().add(jLabel4);
        jLabel4.setBounds(160, 160, 70, 16);

        jLabel5.setForeground(new java.awt.Color(1, 1, 1));
        jLabel5.setText("Passsword");
        getContentPane().add(jLabel5);
        jLabel5.setBounds(160, 230, 70, 16);
        getContentPane().add(inputUsername);
        inputUsername.setBounds(250, 150, 170, 40);

        jLabel6.setFont(new java.awt.Font("Tahoma", 0, 24)); // NOI18N
        jLabel6.setForeground(new java.awt.Color(1, 1, 1));
        jLabel6.setText("Maju Jaya Medical");
        getContentPane().add(jLabel6);
        jLabel6.setBounds(230, 360, 200, 40);

        buttonMasuk.setText("Masuk");
        buttonMasuk.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                buttonMasukActionPerformed(evt);
            }
        });
        getContentPane().add(buttonMasuk);
        buttonMasuk.setBounds(290, 290, 90, 40);
        getContentPane().add(inputPassword);
        inputPassword.setBounds(250, 220, 170, 40);

        jLabel2.setIcon(new javax.swing.ImageIcon(getClass().getResource("/rumahsakit/Logo-PBO-kecil.png"))); // NOI18N
        jLabel2.setText("Logo");
        getContentPane().add(jLabel2);
        jLabel2.setBounds(280, 20, 100, 100);

        jLabel1.setIcon(new javax.swing.ImageIcon(getClass().getResource("/rumahsakit/Login-PBO.jpg"))); // NOI18N
        jLabel1.setText("Background");
        getContentPane().add(jLabel1);
        jLabel1.setBounds(0, 0, 650, 420);

        pack();
    }// </editor-fold>//GEN-END:initComponents

    private void buttonMasukActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_buttonMasukActionPerformed
        // TODO add your handling code here:
        String username = inputUsername.getText();
        String password = new String(inputPassword.getPassword());
        
         try {
            Connection conn=(Connection)koneksi.koneksiDB();
            Statement stt=conn.createStatement();
            sql = "SELECT * FROM user WHERE username='"+username+"' AND password='"+password+"'";
            rs = stt.executeQuery(sql);
            if(rs.next()){
                if(username.equals(rs.getString("username")) && password.equals(rs.getString("password"))){
                    JOptionPane.showMessageDialog(null, "berhasil login");
                    this.dispose();
                    new menuUtama().setVisible(true); 
                }
            }else{
                    JOptionPane.showMessageDialog(null, "username atau password salah");
                }
        } catch (Exception e) {
            JOptionPane.showMessageDialog(this, e.getMessage());
        }
    }//GEN-LAST:event_buttonMasukActionPerformed

    /**
     * @param args the command line arguments
     */
    public static void main(String args[]) {
        /* Set the Nimbus look and feel */
        //<editor-fold defaultstate="collapsed" desc=" Look and feel setting code (optional) ">
        /* If Nimbus (introduced in Java SE 6) is not available, stay with the default look and feel.
         * For details see http://download.oracle.com/javase/tutorial/uiswing/lookandfeel/plaf.html 
         */
        try {
            for (javax.swing.UIManager.LookAndFeelInfo info : javax.swing.UIManager.getInstalledLookAndFeels()) {
                if ("Nimbus".equals(info.getName())) {
                    javax.swing.UIManager.setLookAndFeel(info.getClassName());
                    break;
                }
            }
        } catch (ClassNotFoundException ex) {
            java.util.logging.Logger.getLogger(login.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (InstantiationException ex) {
            java.util.logging.Logger.getLogger(login.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (IllegalAccessException ex) {
            java.util.logging.Logger.getLogger(login.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (javax.swing.UnsupportedLookAndFeelException ex) {
            java.util.logging.Logger.getLogger(login.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        }
        //</editor-fold>

        /* Create and display the form */
        java.awt.EventQueue.invokeLater(new Runnable() {
            public void run() {
                new login().setVisible(true);
            }
        });
    }

    // Variables declaration - do not modify//GEN-BEGIN:variables
    private javax.swing.JButton buttonMasuk;
    private javax.swing.JPasswordField inputPassword;
    private javax.swing.JTextField inputUsername;
    private javax.swing.JLabel jLabel1;
    private javax.swing.JLabel jLabel2;
    private javax.swing.JLabel jLabel4;
    private javax.swing.JLabel jLabel5;
    private javax.swing.JLabel jLabel6;
    // End of variables declaration//GEN-END:variables
}


RumahSakit/src/rumahsakit/menuUtama.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package rumahsakit;

/**
 *
 * @author irfannm
 */
public class menuUtama extends javax.swing.JFrame {

    /**
     * Creates new form menuUtama
     */
    public menuUtama() {
        initComponents();
    }

    /**
     * This method is called from within the constructor to initialize the form.
     * WARNING: Do NOT modify this code. The content of this method is always
     * regenerated by the Form Editor.
     */
    @SuppressWarnings("unchecked")
    // <editor-fold defaultstate="collapsed" desc="Generated Code">//GEN-BEGIN:initComponents
    private void initComponents() {

        jLabel7 = new javax.swing.JLabel();
        jLabel2 = new javax.swing.JLabel();
        buttonPasien = new javax.swing.JButton();
        buttonDokter = new javax.swing.JButton();
        buttonInvoice = new javax.swing.JButton();
        buttonLayanan = new javax.swing.JButton();
        jLabel1 = new javax.swing.JLabel();
        jLabel3 = new javax.swing.JLabel();
        jLabel4 = new javax.swing.JLabel();
        jLabel5 = new javax.swing.JLabel();
        jButton1 = new javax.swing.JButton();
        jLabel8 = new javax.swing.JLabel();

        jLabel7.setIcon(new javax.swing.ImageIcon(getClass().getResource("/rumahsakit/Menu.png"))); // NOI18N
        jLabel7.setText("Background");

        setDefaultCloseOperation(javax.swing.WindowConstants.EXIT_ON_CLOSE);
        setTitle("Maju Jaya Medical");
        setBackground(new java.awt.Color(254, 254, 254));
        getContentPane().setLayout(new org.netbeans.lib.awtextra.AbsoluteLayout());

        jLabel2.setFont(new java.awt.Font("SF Pro Display", 1, 13)); // NOI18N
        jLabel2.setForeground(new java.awt.Color(1, 1, 1));
        jLabel2.setHorizontalAlignment(javax.swing.SwingConstants.CENTER);
        jLabel2.setText("Menu Utama");
        getContentPane().add(jLabel2, new org.netbeans.lib.awtextra.AbsoluteConstraints(300, 10, 80, -1));

        buttonPasien.setText("Data Pasien");
        buttonPasien.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                buttonPasienActionPerformed(evt);
            }
        });
        getContentPane().add(buttonPasien, new org.netbeans.lib.awtextra.AbsoluteConstraints(100, 120, 121, 46));

        buttonDokter.setText("Data Dokter");
        buttonDokter.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                buttonDokterActionPerformed(evt);
            }
        });
        getContentPane().add(buttonDokter, new org.netbeans.lib.awtextra.AbsoluteConstraints(100, 270, 122, 46));

        buttonInvoice.setText("Invoice");
        buttonInvoice.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                buttonInvoiceActionPerformed(evt);
            }
        });
        getContentPane().add(buttonInvoice, new org.netbeans.lib.awtextra.AbsoluteConstraints(460, 110, 93, 46));

        buttonLayanan.setText("Layanan");
        buttonLayanan.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                buttonLayananActionPerformed(evt);
            }
        });
        getContentPane().add(buttonLayanan, new org.netbeans.lib.awtextra.AbsoluteConstraints(470, 270, 90, 46));

        jLabel1.setFont(new java.awt.Font("SF Pro Display", 1, 13)); // NOI18N
        jLabel1.setForeground(new java.awt.Color(1, 1, 1));
        jLabel1.setText("Kelola Data Pasien");
        getContentPane().add(jLabel1, new org.netbeans.lib.awtextra.AbsoluteConstraints(100, 90, -1, -1));

        jLabel3.setFont(new java.awt.Font("SF Pro Display", 1, 13)); // NOI18N
        jLabel3.setForeground(new java.awt.Color(1, 1, 1));
        jLabel3.setText("Kelola Data Dokter");
        getContentPane().add(jLabel3, new org.netbeans.lib.awtextra.AbsoluteConstraints(100, 250, -1, -1));

        jLabel4.setFont(new java.awt.Font("SF Pro Display", 1, 13)); // NOI18N
        jLabel4.setForeground(new java.awt.Color(1, 1, 1));
        jLabel4.setText("Lihat Invoice");
        getContentPane().add(jLabel4, new org.netbeans.lib.awtextra.AbsoluteConstraints(460, 90, -1, -1));

        jLabel5.setFont(new java.awt.Font("SF Pro Display", 1, 13)); // NOI18N
        jLabel5.setForeground(new java.awt.Color(1, 1, 1));
        jLabel5.setText("Layanan");
        getContentPane().add(jLabel5, new org.netbeans.lib.awtextra.AbsoluteConstraints(480, 250, -1, -1));

        jButton1.setText("Log Out");
        jButton1.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                jButton1ActionPerformed(evt);
            }
        });
        getContentPane().add(jButton1, new org.netbeans.lib.awtextra.AbsoluteConstraints(300, 340, -1, -1));

        jLabel8.setIcon(new javax.swing.ImageIcon(getClass().getResource("/rumahsakit/Menu.png"))); // NOI18N
        jLabel8.setText("Background");
        getContentPane().add(jLabel8, new org.netbeans.lib.awtextra.AbsoluteConstraints(0, 0, 700, -1));

        pack();
    }// </editor-fold>//GEN-END:initComponents

    private void buttonLayananActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_buttonLayananActionPerformed
        // TODO add your handling code here:
        this.dispose();
        new Layanan().setVisible(true);
    }//GEN-LAST:event_buttonLayananActionPerformed

    private void buttonPasienActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_buttonPasienActionPerformed
        // TODO add your handling code here:
        this.dispose();
        new DataPasien().setVisible(true);
    }//GEN-LAST:event_buttonPasienActionPerformed

    private void buttonDokterActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_buttonDokterActionPerformed
        // TODO add your handling code here:
        this.dispose();
        new DataDokter().setVisible(true);
    }//GEN-LAST:event_buttonDokterActionPerformed

    private void buttonInvoiceActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_buttonInvoiceActionPerformed
        // TODO add your handling code here:
        this.dispose();
        new Invoice().setVisible(true);
    }//GEN-LAST:event_buttonInvoiceActionPerformed

    private void jButton1ActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_jButton1ActionPerformed
        // TODO add your handling code here:
    this.dispose();
    new login().setVisible(true);
    }//GEN-LAST:event_jButton1ActionPerformed

    /**
     * @param args the command line arguments
     */
    public static void main(String args[]) {
        /* Set the Nimbus look and feel */
        //<editor-fold defaultstate="collapsed" desc=" Look and feel setting code (optional) ">
        /* If Nimbus (introduced in Java SE 6) is not available, stay with the default look and feel.
         * For details see http://download.oracle.com/javase/tutorial/uiswing/lookandfeel/plaf.html 
         */
        try {
            for (javax.swing.UIManager.LookAndFeelInfo info : javax.swing.UIManager.getInstalledLookAndFeels()) {
                if ("Nimbus".equals(info.getName())) {
                    javax.swing.UIManager.setLookAndFeel(info.getClassName());
                    break;
                }
            }
        } catch (ClassNotFoundException ex) {
            java.util.logging.Logger.getLogger(menuUtama.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (InstantiationException ex) {
            java.util.logging.Logger.getLogger(menuUtama.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (IllegalAccessException ex) {
            java.util.logging.Logger.getLogger(menuUtama.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (javax.swing.UnsupportedLookAndFeelException ex) {
            java.util.logging.Logger.getLogger(menuUtama.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        }
        //</editor-fold>

        /* Create and display the form */
        java.awt.EventQueue.invokeLater(new Runnable() {
            public void run() {
                new menuUtama().setVisible(true);
            }
        });
    }

    // Variables declaration - do not modify//GEN-BEGIN:variables
    private javax.swing.JButton buttonDokter;
    private javax.swing.JButton buttonInvoice;
    private javax.swing.JButton buttonLayanan;
    private javax.swing.JButton buttonPasien;
    private javax.swing.JButton jButton1;
    private javax.swing.JLabel jLabel1;
    private javax.swing.JLabel jLabel2;
    private javax.swing.JLabel jLabel3;
    private javax.swing.JLabel jLabel4;
    private javax.swing.JLabel jLabel5;
    private javax.swing.JLabel jLabel7;
    private javax.swing.JLabel jLabel8;
    // End of variables declaration//GEN-END:variables
}


RumahSakit/src/rumahsakit/Invoice.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package rumahsakit;
import java.sql.*;
import javax.swing.JOptionPane;
import javax.swing.table.DefaultTableModel;
import java.util.Scanner;
import java.awt.*;
   import java.io.FileOutputStream;
    import com.itextpdf.text.Document;
    import com.itextpdf.text.Paragraph;
    import com.itextpdf.text.Phrase;
    import com.itextpdf.text.pdf.PdfWriter;
    import com.itextpdf.text.pdf.PdfPTable;
    import com.itextpdf.text.pdf.PdfPCell;
    import com.itextpdf.text.Image;
import static rumahsakit.generatePDF.printpdf;
/**
 *
 * @author X230
 */
public class Invoice extends javax.swing.JFrame {
DefaultTableModel tabModel;
ResultSet rsinvoice=null;
    /**
     * Creates new form Invoice
     */
    public Invoice() {
        initComponents();
        
    }
        private void tampilData(String number_input){
        try{
            Object[] judul_kolom = {"id_layanan", "id_pasien", "id_dokter", "jenis", "keterangan", "waktu", "harga"};
            tabModel=new DefaultTableModel(null,judul_kolom);
            TableInvoice.setModel(tabModel);
            
            Connection conn=(Connection)koneksi.koneksiDB();
            Statement stt=conn.createStatement();
            tabModel.getDataVector().removeAllElements();
            rsinvoice=stt.executeQuery("SELECT * from layanan where id_pasien = '" + number_input + "' ");  
            while(rsinvoice.next()){
                Object[] data={
                    rsinvoice.getString("id_layanan"),
                    rsinvoice.getString("id_pasien"),
                    rsinvoice.getString("id_dokter"),
                    rsinvoice.getString("jenis"),
                    rsinvoice.getString("keterangan"),
                    rsinvoice.getString("waktu"),
                    rsinvoice.getString("harga")
                };
            tabModel.addRow(data);
            }                
        } catch (Exception ex) {
        System.err.println(ex.getMessage());
        }
        
    }

        private void TotalHarga(String number_input){
        try{
            Object[] judul_kolom = {"id_layanan", "id_pasien", "id_dokter", "jenis", "keterangan", "waktu", "harga"};            
            Connection conn=(Connection)koneksi.koneksiDB();
            Statement stt=conn.createStatement();
            rsinvoice=stt.executeQuery("SELECT SUM(harga) as total from layanan where id_pasien = '" + number_input + "' ");  
            while(rsinvoice.next()){
                String data = rsinvoice.getString("total");
                labelhasil.setText("Rp "+data+",-");
            }                
        } catch (Exception ex) {
        System.err.println(ex.getMessage());
        }
        
    }
            public static void printpdf(String idpelanggan){
    try{
    String file_name="..\\InvoicePasien'"+ idpelanggan +"'.pdf";
    Document document=new Document();
    PdfWriter.getInstance(document, new FileOutputStream(file_name));
    document.open();
    String img = "https://cdn.discordapp.com/attachments/839027400166735932/851501331922681887/kop.png";
    document.add(Image.getInstance( img ));
    Connection connection=(Connection)koneksi.koneksiDB();
    ResultSet rs=null;
    PreparedStatement ps=null;

    String query = "select * from layanan where id_pasien = '"+idpelanggan +"'";
    ps=connection.prepareStatement(query);
    rs=ps.executeQuery();

    PdfPTable table = new PdfPTable(5);
    PdfPCell cl = new PdfPCell(new Phrase("Waktu"));
    table.addCell(cl);
    cl = new PdfPCell(new Phrase("Id_Pasien"));
    table.addCell(cl);
    cl = new PdfPCell(new Phrase("Jenis Layanan"));
    table.addCell(cl);
    cl = new PdfPCell(new Phrase("Keterangan"));
    table.addCell(cl);
    cl = new PdfPCell(new Phrase("Biaya"));
    table.addCell(cl);
    table.setHeaderRows(1);
    document.add(new Paragraph("Rincian:"));
    document.add(new Paragraph(" "));
    while(rs.next()){
    table.addCell(rs.getString("waktu"));
    table.addCell(rs.getString("id_pasien"));
    table.addCell(rs.getString("jenis"));
    table.addCell(rs.getString("keterangan"));
    table.addCell(rs.getString("harga"));
    }    
    ResultSet rsinvoice=null;
    Connection conn=(Connection)koneksi.koneksiDB();
    Statement stt=conn.createStatement();
    rsinvoice=stt.executeQuery("SELECT SUM(harga) as total from layanan where   id_pasien = '"+idpelanggan +"'");  

         
    System.out.println("Finish");
    document.add(table);
    
    document.add(new Paragraph(" "));
        while(rsinvoice.next()){
        String data = rsinvoice.getString("total");
        document.add(new Paragraph("Total biaya yang harus dibayar sebesar: Rp "+data+",-")); 
        document.add(new Paragraph(""));
        document.add(new Paragraph(" "));
        document.add(new Paragraph(" "));
        }     
    document.close();
}
catch(Exception e){
System.err.println(e);
}    
        
        
    };

    /**
     * This method is called from within the constructor to initialize the form.
     * WARNING: Do NOT modify this code. The content of this method is always
     * regenerated by the Form Editor.
     */
    @SuppressWarnings("unchecked")
    // <editor-fold defaultstate="collapsed" desc="Generated Code">//GEN-BEGIN:initComponents
    private void initComponents() {

        jScrollPane1 = new javax.swing.JScrollPane();
        TableInvoice = new javax.swing.JTable();
        jLabel1 = new javax.swing.JLabel();
        jLabel2 = new javax.swing.JLabel();
        jTextFieldnim = new javax.swing.JTextField();
        jLabel3 = new javax.swing.JLabel();
        labelhasil = new javax.swing.JLabel();
        jLabel5 = new javax.swing.JLabel();
        button2 = new java.awt.Button();
        button1 = new java.awt.Button();
        button3 = new java.awt.Button();
        button4 = new java.awt.Button();
        jLabel4 = new javax.swing.JLabel();

        setDefaultCloseOperation(javax.swing.WindowConstants.EXIT_ON_CLOSE);
        setTitle("Maju Jaya Medical");
        setMinimumSize(new java.awt.Dimension(915, 560));
        getContentPane().setLayout(null);

        TableInvoice.setModel(new javax.swing.table.DefaultTableModel(
            new Object [][] {
                {null, null, null, null, null, null, null},
                {null, null, null, null, null, null, null},
                {null, null, null, null, null, null, null},
                {null, null, null, null, null, null, null}
            },
            new String [] {
                "id_layanan", "id_pasien", "id_dokter", "jenis", "keterangan", "waktu", "harga"
            }
        ));
        jScrollPane1.setViewportView(TableInvoice);

        getContentPane().add(jScrollPane1);
        jScrollPane1.setBounds(420, 77, 452, 402);

        jLabel1.setFont(new java.awt.Font("Dialog", 0, 36)); // NOI18N
        jLabel1.setText("Invoice");
        getContentPane().add(jLabel1);
        jLabel1.setBounds(378, 12, 110, 47);

        jLabel2.setFont(new java.awt.Font("Dialog", 0, 24)); // NOI18N
        jLabel2.setText("Masukan Id_Pasien");
        getContentPane().add(jLabel2);
        jLabel2.setBounds(20, 100, 220, 20);

        jTextFieldnim.setText(" ");
        jTextFieldnim.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                jTextFieldnimActionPerformed(evt);
            }
        });
        getContentPane().add(jTextFieldnim);
        jTextFieldnim.setBounds(260, 100, 133, 20);

        jLabel3.setFont(new java.awt.Font("Dialog", 0, 24)); // NOI18N
        jLabel3.setText("Total Biaya");
        getContentPane().add(jLabel3);
        jLabel3.setBounds(45, 173, 119, 32);

        labelhasil.setFont(new java.awt.Font("Dialog", 1, 18)); // NOI18N
        labelhasil.setText("Rp 0,-");
        getContentPane().add(labelhasil);
        labelhasil.setBounds(260, 170, 133, 32);

        jLabel5.setIcon(new javax.swing.ImageIcon(getClass().getResource("/rumahsakit/Logo-PBO-kecil-bgt.png"))); // NOI18N
        jLabel5.setText("Logo");
        getContentPane().add(jLabel5);
        jLabel5.setBounds(10, 10, 30, 30);

        button2.setLabel("Clear");
        button2.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                button2ActionPerformed(evt);
            }
        });
        getContentPane().add(button2);
        button2.setBounds(150, 290, 90, 50);

        button1.setLabel("GET");
        button1.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                button1ActionPerformed(evt);
            }
        });
        getContentPane().add(button1);
        button1.setBounds(30, 290, 90, 50);

        button3.setLabel("Print Invoice");
        button3.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                button3ActionPerformed(evt);
            }
        });
        getContentPane().add(button3);
        button3.setBounds(280, 290, 90, 50);

        button4.setLabel("Kembali");
        button4.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                button4ActionPerformed(evt);
            }
        });
        getContentPane().add(button4);
        button4.setBounds(40, 440, 60, 30);

        jLabel4.setIcon(new javax.swing.ImageIcon(getClass().getResource("/rumahsakit/Bg-PBO-coab.jpg"))); // NOI18N
        jLabel4.setText("Background");
        getContentPane().add(jLabel4);
        jLabel4.setBounds(0, 0, 900, 520);

        pack();
    }// </editor-fold>//GEN-END:initComponents

    private void jTextFieldnimActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_jTextFieldnimActionPerformed
        // TODO add your handling code here:
    }//GEN-LAST:event_jTextFieldnimActionPerformed

    private void button1ActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_button1ActionPerformed
        // TODO add your handling code here:
       String a = jTextFieldnim.getText();

        tampilData(a);
        TotalHarga(a);

    }//GEN-LAST:event_button1ActionPerformed

    private void button2ActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_button2ActionPerformed
        // TODO add your handling code here:
                jTextFieldnim.setText("");
        labelhasil.setText("Rp 0.00,-");
        tampilData(null);;
    }//GEN-LAST:event_button2ActionPerformed

    private void button3ActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_button3ActionPerformed
        // TODO add your handling code here:
        String a = jTextFieldnim.getText();
        printpdf(a);
        JOptionPane.showMessageDialog(null,"File PDF sudah tergenerate");
    }//GEN-LAST:event_button3ActionPerformed

    private void button4ActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_button4ActionPerformed
        // TODO add your handling code here:
        this.dispose();
        new menuUtama().setVisible(true);
    }//GEN-LAST:event_button4ActionPerformed

    /**
     * @param args the command line arguments
     */
    public static void main(String args[]) {
        /* Set the Nimbus look and feel */
        //<editor-fold defaultstate="collapsed" desc=" Look and feel setting code (optional) ">
        /* If Nimbus (introduced in Java SE 6) is not available, stay with the default look and feel.
         * For details see http://download.oracle.com/javase/tutorial/uiswing/lookandfeel/plaf.html 
         */
        try {
            for (javax.swing.UIManager.LookAndFeelInfo info : javax.swing.UIManager.getInstalledLookAndFeels()) {
                if ("Nimbus".equals(info.getName())) {
                    javax.swing.UIManager.setLookAndFeel(info.getClassName());
                    break;
                }
            }
        } catch (ClassNotFoundException ex) {
            java.util.logging.Logger.getLogger(Invoice.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (InstantiationException ex) {
            java.util.logging.Logger.getLogger(Invoice.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (IllegalAccessException ex) {
            java.util.logging.Logger.getLogger(Invoice.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (javax.swing.UnsupportedLookAndFeelException ex) {
            java.util.logging.Logger.getLogger(Invoice.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        }
        //</editor-fold>

        /* Create and display the form */
        java.awt.EventQueue.invokeLater(new Runnable() {
            public void run() {
                new Invoice().setVisible(true);
            }
        });
    }
    // Variables declaration - do not modify//GEN-BEGIN:variables
    private javax.swing.JTable TableInvoice;
    private java.awt.Button button1;
    private java.awt.Button button2;
    private java.awt.Button button3;
    private java.awt.Button button4;
    private javax.swing.JLabel jLabel1;
    private javax.swing.JLabel jLabel2;
    private javax.swing.JLabel jLabel3;
    private javax.swing.JLabel jLabel4;
    private javax.swing.JLabel jLabel5;
    private javax.swing.JScrollPane jScrollPane1;
    private javax.swing.JTextField jTextFieldnim;
    private javax.swing.JLabel labelhasil;
    // End of variables declaration//GEN-END:variables
}
